scripts/analysis/compare_grid_detection.py

```
@@ -0,0 +1,436 @@
+def detect_grid_brightness_method(image_path, min_grid_size=100, max_grid_size=150):
+    """
+    Brightness-based grid detection method using analyze_brightness.py
+    """
+    try:
+        # Run the analyze_brightness script and capture output
+        import os
+
+        env = os.environ.copy()
+        env["PYTHONIOENCODING"] = "utf-8"
+
+        result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)